app/src/main/java/org/mozilla/fenix/perf/PerfNavController.kt

```diff
@@ -0,0 +1,37 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.perf
+
+import androidx.annotation.VisibleForTesting
+import androidx.navigation.NavController
+import kotlinx.coroutines.Job
+import java.util.WeakHashMap
+
+/**
+ * We need to keep a weak reference to our NavController in case our activity
+ * gets re-created and we need to link our NavGraph to the NavController again.
+ * The value of this map is the job associated to the NavController which should link the NavGraph
+ * to our NavController.
+ */
+@VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+val map = WeakHashMap<NavController, Job>()
+
+fun addNavToMap(navController: NavController, job: Job) {
+    map[navController] = job
+}
+
+/**
+ * The job should block the main thread if it isn't completed so that the NavGraph can be loaded
+ * before any navigation is done.
+ */
+fun waitForNavGraphInflation(navController: NavController) {
+    map.getValue(navController).also {
```

Comment:
If the navController is missing (which is unexpected), we'll throw `java.util.NoSuchElementException: Key NavController(navController#32) is missing in the map.`. To someone unfamiliar with this code, it's unintuitive. The underlying problem (we think) is that we didn't actually start the async nav graph inflation. Can we catch this error before `map.getValue()` throwsn and make it more explicit?